File: app/src/main/java/com/unho/unhomeals/MainActivity.java
package com.unho.unhomeals;

import android.annotation.SuppressLint;
import android.content.Intent;
import android.graphics.Bitmap;
import android.os.Bundle;
import android.text.Editable;
import android.text.TextWatcher;
import android.util.Log;
import android.view.View;
import android.widget.ImageView;
import android.widget.RatingBar;
import android.widget.TextView;

import androidx.appcompat.app.AppCompatActivity;
import androidx.viewpager2.widget.CompositePageTransformer;
import androidx.viewpager2.widget.ViewPager2;

import com.google.zxing.BarcodeFormat;
import com.google.zxing.WriterException;
import com.journeyapps.barcodescanner.BarcodeEncoder;
import com.unho.unhomeals.adapter.MainAdapter;
import com.unho.unhomeals.data.SharedPf;

import java.text.ParseException;
import java.text.SimpleDateFormat;
import java.util.Calendar;
import java.util.GregorianCalendar;
import java.util.Map;
import java.util.Objects;

public class MainActivity extends AppCompatActivity {

    static Map<String, String> dietInfo;
    MainAdapter adapter;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        adapter = IntroActivity.adapter;
        dietInfo = IntroActivity.dietInfo;

        ImageView iv_profile = findViewById(R.id.iv_profile);
        TextView tv_profile = findViewById(R.id.tv_profile);

        String UID = SharedPf.getString(this, "UID");

        if (UID.equals("0")) {
            iv_profile.setImageResource(R.drawable.pf_teacher);
            tv_profile.setText("선생님");

            float rating = 4.5f; // TODO SERVER : Get the average of ratings from server
            rating(rating);
        } else {
            iv_profile.setImageResource(R.drawable.pf_student);
            String[] name = IntroActivity.userInfo.get(UID).split(" ");
            tv_profile.setText(String.format("%c-%c %s", name[0].charAt(0), name[0].charAt(2), name[1]));
            barcode(UID);
        }

        // Date
        TextView tv_today = findViewById(R.id.tv_today);
        tv_today.addTextChangedListener(new TextWatcher() {
            @Override
            public void beforeTextChanged(CharSequence charSequence, int i, int i1, int i2) {}

            @Override
            public void onTextChanged(CharSequence charSequence, int i, int i1, int i2) {
                @SuppressLint("SimpleDateFormat")
                SimpleDateFormat dateFormat1 = new SimpleDateFormat("yyyy. M. d.");
                @SuppressLint("SimpleDateFormat")
                SimpleDateFormat dateFormat2 = new SimpleDateFormat("yyyyMMdd");

                try {
                    String today = dateFormat2.format(Objects.requireNonNull(dateFormat1.parse(charSequence.toString())));

                    String[] contents = {
                            dietInfo.get(today+"조식"),
                            dietInfo.get(today+"중식"),
                            dietInfo.get(today+"석식"),
                    };

                    for (int j = 0; j < contents.length; j++) {
                        if (contents[j] == null) contents[j] = "급식 정보가 없습니다.";
                        adapter.getItemAt(j).setContent(contents[j]);
                        adapter.notifyItemChanged(j);
                    }
                } catch (ParseException ignored) {}
            }

            @Override
            public void afterTextChanged(Editable editable) {}
        });

        Calendar calendar = new GregorianCalendar();

        @SuppressLint("SimpleDateFormat")
        SimpleDateFormat format = new SimpleDateFormat("yyyy. M. d.");
        tv_today.setText(format.format(calendar.getTime()));
        tv_today.setOnClickListener(view -> tv_today.setText(format.format(new GregorianCalendar().getTime())));

        findViewById(R.id.btn_prev).setOnClickListener(view -> {
            calendar.add(Calendar.DATE, -1);
            tv_today.setText(format.format(calendar.getTime()));
        });

        findViewById(R.id.btn_next).setOnClickListener(view -> {
            calendar.add(Calendar.DATE, 1);
            tv_today.setText(format.format(calendar.getTime()));
        });

        // ViewPager2
        CompositePageTransformer transformer = new CompositePageTransformer();
        transformer.addTransformer((page, position) -> page.setScaleY(0.8f+(1-Math.abs(position))*0.2f));

        ViewPager2 vp_meals = findViewById(R.id.vp_meals);
        vp_meals.setAdapter(adapter);
        vp_meals.setOffscreenPageLimit(3);
        vp_meals.setPageTransformer(transformer);

        int hour = calendar.get(Calendar.HOUR_OF_DAY);

        if (hour < 8) {
            vp_meals.setCurrentItem(0);
        } else if (hour < 14) {
            vp_meals.setCurrentItem(1);
        } else {
            vp_meals.setCurrentItem(2);
        }

        // Buttons
        findViewById(R.id.btn_apply).setOnClickListener(view -> {
            Intent intent = new Intent(this, ApplyActivity.class);
            startActivity(intent);
        });

        findViewById(R.id.btn_rate).setOnClickListener(view -> {
            Intent intent = new Intent(this, RateActivity.class);
            startActivity(intent);
        });

        findViewById(R.id.btn_ranking).setOnClickListener(view -> {
            Intent intent = new Intent(this, RankingActivity.class);
            startActivity(intent);
        });

        findViewById(R.id.btn_setting).setOnClickListener(view -> {
            Intent intent = new Intent(this, SettingActivity.class);
            startActivity(intent);
        });
    }

    private void barcode(String id) {
        ImageView iv_barcode = findViewById(R.id.iv_barcode);
        TextView tv_barcode = findViewById(R.id.tv_barcode);

        iv_barcode.setVisibility(View.VISIBLE);
        tv_barcode.setVisibility(View.VISIBLE);

        try {
            Bitmap bitmap = new BarcodeEncoder().encodeBitmap(id, BarcodeFormat.CODE_39, 800, 150);
            iv_barcode.setImageBitmap(bitmap);
            tv_barcode.setText(id);
        } catch(WriterException ignored) {}
    }

    private void rating(float rating) {
        RatingBar rb_rating = findViewById(R.id.rb_rating);
        TextView tv_rating = findViewById(R.id.tv_rating);

        rb_rating.setVisibility(View.VISIBLE);
        tv_rating.setVisibility(View.VISIBLE);

        String text;

        if (rating < 1.5) {
            text = "만족도 매우 낮음";
        } else if (rating < 2.5) {
            text = "만족도 낮음";
        } else if (rating < 3.5) {
            text = "만족도 보통";
        } else if (rating < 4.5) {
            text = "만족도 높음";
        } else {
            text = "만족도 매우 높음";
        }

        rb_rating.setRating(rating);
        tv_rating.setText(text);
    }
}

File: app/src/main/java/com/smoodi/smoodi/MainActivity.kt
package com.smoodi.smoodi

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button
import android.widget.ImageView
import android.widget.TextView
import androidx.core.content.ContextCompat
import androidx.core.content.res.ResourcesCompat
import androidx.core.widget.addTextChangedListener
import com.google.zxing.BarcodeFormat
import com.journeyapps.barcodescanner.BarcodeEncoder
import com.smoodi.smoodi.data.Data
import java.text.SimpleDateFormat
import java.util.Locale

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        /* Setup Logout Button */
        val tvReset: TextView = findViewById(R.id.tvReset)
        tvReset.setOnClickListener {

            Data().setUserData(this, "", "")

            val intent = Intent(this, LoginActivity::class.java)
            finish()
            startActivity(intent)
        }

        /* Setup Views */
        setupBarcode()
        setupDate()
        setupMeal()
        setupMenu()

        val tvIcon: TextView = findViewById(R.id.tvIcon)
        val tvName: TextView = findViewById(R.id.tvName)

        if (Data.isSuper()) {
            val gold = ContextCompat.getColor(this, R.color.gold)
            tvIcon.setTextColor(gold)
        }

        tvName.text = Data.name

        // To display open source notices
//        startActivity(Intent(this, OssLicensesMenuActivity::class.java))

    }

    private fun setupBarcode() {

        val ivBarcode: ImageView = findViewById(R.id.ivBarcode)
        val tvBarcode: TextView = findViewById(R.id.tvBarcode)

        val barcodeEncoder = BarcodeEncoder()
        val bitmap = barcodeEncoder.encodeBitmap(
            Data.id, BarcodeFormat.CODE_128, 600, 200)

        ivBarcode.setImageBitmap(bitmap)
        tvBarcode.text = Data.id

    }

    private fun setupDate() {

        val tvDate: TextView = findViewById(R.id.tvDate)

        val dateFormat1 = SimpleDateFormat("yyyyMMdd", Locale.KOREA)
        val dateFormat2 = SimpleDateFormat("yyyy. M. d.", Locale.KOREA)

        val date = dateFormat2.format(dateFormat1.parse(Data.now)!!)

        tvDate.text = date

    }

    private fun setupMeal() {

        val tvMeal1: TextView = findViewById(R.id.tvMeal1)
        val tvMeal2: TextView = findViewById(R.id.tvMeal2)
        val tvMeal3: TextView = findViewById(R.id.tvMeal3)

        val tvMeal: TextView = findViewById(R.id.tvMeal)
        tvMeal.addTextChangedListener { text ->
            if (text.toString().isEmpty()) {
                tvMeal.text = getString(R.string.main_meal_null)
            }
        }

        val fontBlack = ResourcesCompat.getFont(this, R.font.black)
        val fontRegular = ResourcesCompat.getFont(this, R.font.regular)

        tvMeal1.setOnClickListener {
            tvMeal1.typeface = fontBlack
            tvMeal2.typeface = fontRegular
            tvMeal3.typeface = fontRegular

            tvMeal.text = Data.meals[getString(R.string.main_meal1)]
        }
        tvMeal2.setOnClickListener {
            tvMeal1.typeface = fontRegular
            tvMeal2.typeface = fontBlack
            tvMeal3.typeface = fontRegular

            tvMeal.text = Data.meals[getString(R.string.main_meal2)]
        }.also { tvMeal2.callOnClick() }
        tvMeal3.setOnClickListener {
            tvMeal1.typeface = fontRegular
            tvMeal2.typeface = fontRegular
            tvMeal3.typeface = fontBlack

            tvMeal.text = Data.meals[getString(R.string.main_meal3)]
        }
    }

    private fun setupMenu() {

        val btGoRank: Button = findViewById(R.id.btGoRank)
        val btGoRate: Button = findViewById(R.id.btGoRate)
        val btGoApply: Button = findViewById(R.id.btGoApply)

        btGoRank.setOnClickListener {
            val intent = Intent(this, RankActivity::class.java)
            startActivity(intent)
        }
        btGoRate.setOnClickListener {
            val intent = Intent(this, RateActivity::class.java)
            startActivity(intent)
        }
        btGoApply.setOnClickListener {
            val intent = Intent(this, ApplyActivity::class.java)
            startActivity(intent)
        }
    }
}